packages/react-components/react-positioning/src/utils/writeArrowUpdates.ts

```diff
@@ -1,18 +1,32 @@
-import { MiddlewareData } from '@floating-ui/dom';
+import { MiddlewareData, Placement } from '@floating-ui/dom';
 
 /**
  * Writes all DOM element updates after position is computed
  */
-export function writeArrowUpdates(options: { arrow: HTMLElement | null; middlewareData: MiddlewareData }) {
-  const { arrow, middlewareData } = options;
+export function writeArrowUpdates(options: {
+  arrow: HTMLElement | null;
+  middlewareData: MiddlewareData;
+  placement: Placement;
+}) {
+  const { arrow, middlewareData, placement } = options;
   if (!middlewareData.arrow || !arrow) {
     return;
   }
 
   const { x: arrowX, y: arrowY } = middlewareData.arrow;
 
+  const staticSide = {
+    top: 'bottom',
+    right: 'left',
+    bottom: 'top',
+    left: 'right',
+  }[placement.split('-')[0]] as string;
+
   Object.assign(arrow.style, {
-    left: `${arrowX}px`,
-    top: `${arrowY}px`,
+    left: arrowX !== null && arrowX !== undefined ? `${arrowX}px` : '',
+    top: arrowY !== null && arrowY !== undefined ? `${arrowY}px` : '',
+    right: '',
+    bottom: '',
+    [staticSide]: '-5px',
```

Comment:
What does this magic number?

Reply:
I thought that's needed (as it's shown in floating ui docs) and during implementation the arrow was not visible for top examples, I guess I had something else changed, removed now and it works fine without, VR tests also should be happier 🙂

Reply:
@mainframev I see, so the root issue was:

<img width="488" alt="image" src="https://github.com/user-attachments/assets/dd56525f-dc1c-4eac-9f30-ea2f456b13de">

Reply:
Yes, exactly